src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -561,31 +562,37 @@ public async ValueTask<HttpResponseMessage> SendWithVersionDetectionAndRetryAsyn
             }
         }
 
-        private async ValueTask<(Stream, TransportContext?, IPEndPoint?)> ConnectAsync(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
+        private async ValueTask<(Stream, TransportContext?, Activity?, IPEndPoint?)> ConnectAsync(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
         {
             Stream? stream = null;
             IPEndPoint? remoteEndPoint = null;
+
+            // Connection activities should be new roots and not parented under whatever
+            // request happens to be in progress when the connection is started.
```

Comment:
Why is this? What goes wrong if we don't do this?

Reply:
The connection attempt spawned by a request may not be the connection serving the request, thus creating a parent-child relationship would be unfortunate.

Instead, we decided to use activity links to link the connection that is actually serving the request at the moment a connection is associated to a request from the pool.

Reply:
Also, even if we ignore or change that implementation detail:

For requests, N requests share 1 connection. However Activities are a tree, 1 connection activity can not have N distinct parents.